server/graphql/v1/mutations/orders.js

```diff
@@ -158,20 +158,69 @@ const checkGuestContribution = async (order, loaders) => {
     throw new BadRequest('You need to provide a guest profile with an email for logged out contributions');
   } else if (!guestInfo.email || !isEmail(guestInfo.email)) {
     throw new BadRequest('You need to provide a valid email');
-  } else if (order.totalAmount > 25000) {
-    const location = guestInfo.location || {};
-    if (!guestInfo.name) {
-      throw new BadRequest('Contributions that are more than $250 must have a name attached');
-    } else if (order.totalAmount > 500000 && !location.structured && (!location.address || !location.country)) {
-      throw new BadRequest('Contributions that are more than $5000 must have an address attached');
-    }
   } else if (order.fromCollective) {
     throw new BadRequest('You need to be logged in to specify a contributing profile');
   } else if (order.paymentMethod?.id || order.paymentMethod?.uuid) {
     throw new BadRequest('You need to be logged in to be able to use an existing payment method');
   }
 };
 
+const mustUpdateLocation = (existingLocation, newLocation) => {
+  const simpleFields = ['country', 'name', 'address'];
+  const hasUpdatedSimpleField = field => newLocation[field] && newLocation[field] !== existingLocation[field];
+  return simpleFields.some(hasUpdatedSimpleField) || !isEqual(existingLocation.structured, newLocation.structured);
+};
+
+const mustUpdateNames = (fromAccount, fromAccountInfo) => {
+  return (!fromAccount.name && fromAccountInfo?.name) || (!fromAccount.legalName && fromAccountInfo?.legalName);
+};
+
+/**
+ * Checks that the profile has all requirements for this contribution (name, address, etc) and updates
+ * it if necessary. Must be called **after** checking permissions and authentication!
+ */
+const checkAndUpdateProfileInfo = async (order, fromAccount, isGuest) => {
+  const { totalAmount, fromAccountInfo, guestInfo } = order;
+  const accountUpdatePayload = {};
+  const location = fromAccountInfo?.location || guestInfo?.location || fromAccount.location;
+
+  // Only enforce profile checks for guests at the moment
+  if (isGuest) {
+    // Contributions that are more than $5000 must have an address attached
+    if (totalAmount > 5000e2) {
```

Comment:
I like the cents notation, but we should also verify the currency. 👍🏼

Reply:
Damn that a very good point, we're not considering currency at any point here.

Reply:
I opened a followup issue for that [here](https://github.com/opencollective/opencollective/issues/5092), and I'm going to ignore this check if the currency is not USD for now.